src/main/java/pl/edu/agh/firecell/engine/algorithm/FirePropagator.java

```diff
@@ -0,0 +1,103 @@
+package pl.edu.agh.firecell.engine.algorithm;
+
+import org.joml.Vector3i;
+import pl.edu.agh.firecell.model.Cell;
+import pl.edu.agh.firecell.model.State;
+import pl.edu.agh.firecell.model.util.NeighbourUtils;
+
+import java.util.Collections;
+import java.util.NoSuchElementException;
+import java.util.stream.Stream;
+
+import static pl.edu.agh.firecell.engine.algorithm.BasicAlgorithm.*;
+import static pl.edu.agh.firecell.model.Material.AIR;
+import static pl.edu.agh.firecell.model.Material.WOOD;
+
+public class FirePropagator {
```

Comment:
Sort private methods topologically.
It means that if one method is used by another, it should go beneath it.
In that case reverse them totally XD.